File: questModule/src/main/java/ru/nekit/android/qls/quest/answer/SimpleExampleAnswerChecker.java
package ru.nekit.android.qls.quest.answer;

import android.support.annotation.NonNull;

import ru.nekit.android.qls.quest.answer.shared.QuestAnswerChecker;
import ru.nekit.android.qls.quest.base.Quest;
import ru.nekit.android.qls.quest.math.MathematicalOperation;
import ru.nekit.android.qls.quest.math.MathematicalSignComparison;
import ru.nekit.android.qls.quest.types.NumberSummandQuest;

import static ru.nekit.android.qls.quest.math.MathematicalOperation.ADDITION;
import static ru.nekit.android.qls.quest.math.MathematicalOperation.SUBTRACTION;

public class SimpleExampleAnswerChecker extends QuestAnswerChecker<Object> {

    @Override
    public boolean checkAlternativeInput(@NonNull Quest quest, @NonNull Object answer) {
        NumberSummandQuest numberSummandQuest = (NumberSummandQuest) quest;
        boolean result = false;
        switch (quest.getQuestionType()) {

            case COMPARISON:

                MathematicalSignComparison sign = (MathematicalSignComparison) answer;
                result = numberSummandQuest.getSign() == sign;

                break;

            case UNKNOWN_OPERATION:

                MathematicalOperation operation = ADDITION;
                MathematicalOperation answerOperation = (MathematicalOperation) answer;
                if (numberSummandQuest.leftNode[numberSummandQuest.unknownOperatorIndex + 1] < 0) {
                    operation = SUBTRACTION;
                }
                result = operation == answerOperation;

                break;

        }

        return result;
    }
}

File: questDomain/src/main/java/ru/nekit/android/qls/domain/answerChecker/SimpleExampleAnswerChecker.kt
package ru.nekit.android.qls.domain.answerChecker

import ru.nekit.android.qls.domain.answerChecker.common.QuestAnswerChecker
import ru.nekit.android.qls.domain.model.math.MathematicalOperation
import ru.nekit.android.qls.domain.model.math.MathematicalOperation.ADDITION
import ru.nekit.android.qls.domain.model.math.MathematicalOperation.SUBTRACTION
import ru.nekit.android.qls.domain.model.math.MathematicalSignComparison
import ru.nekit.android.qls.domain.model.quest.NumberSummandQuest
import ru.nekit.android.qls.shared.model.QuestionType.COMPARISON
import ru.nekit.android.qls.shared.model.QuestionType.UNKNOWN_OPERATION

class SimpleExampleAnswerChecker : QuestAnswerChecker<NumberSummandQuest>() {

    override fun checkAlternativeInput(quest: NumberSummandQuest, answer: Any): Boolean {
        return when (quest.questionType) {

            COMPARISON -> {
                val sign = answer as MathematicalSignComparison
                quest.sign == sign
            }

            UNKNOWN_OPERATION -> {
                var operation = ADDITION
                val answerOperation = answer as MathematicalOperation
                if (quest.leftNode[quest.unknownOperatorIndex + 1] < 0) {
                    operation = SUBTRACTION
                }
                operation == answerOperation
            }
            else -> false
        }
    }
}
